chronos/asyncloop.nim

```diff
@@ -1045,7 +1041,10 @@ proc wait*[T](fut: Future[T], timeout = InfiniteDuration): Future[T] =
       when T is void:
         retFuture.complete()
       else:
-        retFuture.complete(fut.read())
+        try:
+          retFuture.complete(fut.read())
+        except CatchableError as exc:
+          retFuture.fail(exc)
```

Comment:
This should be `Assertion` or `Defect` error, because situation that `fut.read()` will raise is impossible.